src/main/webapp/app/exercises/programming/manage/grading/charts/test-case-distribution-chart.component.ts

```diff
@@ -74,50 +143,73 @@ export class TestCaseDistributionChartComponent implements OnChanges {
             };
         });
 
-        if (this.weightChartDatasets.length !== testCaseScores.length) {
+        if (this.ngxWeightData[0].length !== testCaseScores.length) {
             const testCaseColors = {};
 
-            this.weightChartDatasets = [];
-            this.pointsChartDatasets = [];
+            this.ngxWeightData = [];
+            this.ngxPointsData = [];
+
+            const weight = { name: this.translateService.instant('artemisApp.programmingExercise.configureGrading.charts.testCaseWeights.weight'), series: [] as any[] };
+            const weightAndBonus = {
+                name: this.translateService.instant('artemisApp.programmingExercise.configureGrading.charts.testCaseWeights.weightAndBonus'),
+                series: [] as any[],
+            };
+
+            const points = { name: this.translateService.instant('artemisApp.programmingExercise.configureGrading.charts.testCasePoints.points'), series: [] as any[] };
 
             for (let i = 0; i < testCaseScores.length; i++) {
                 const element = testCaseScores[i];
 
                 const label = element.label;
-                const backgroundColor = this.getColor(+i / this.testCases.length, 50);
-                const hoverBackgroundColor = this.getColor(+i / this.testCases.length, 60);
-
-                testCaseColors[label] = backgroundColor;
-
-                this.weightChartDatasets.push({
-                    label,
-                    backgroundColor,
-                    hoverBackgroundColor,
-                    data: [element.relWeight, element.relScore],
-                });
-
-                this.pointsChartDatasets.push({
-                    label,
-                    backgroundColor,
-                    hoverBackgroundColor,
-                    data: [element.relPoints],
-                });
+                const color = this.getColor(+i / this.testCases.length, 50);
+
+                weight.series.push({ name: element.label, value: element.relWeight, bonus: element.relScore });
+                weightAndBonus.series.push({ name: element.label, value: element.relScore, weight: element.relWeight });
+
+                points.series.push({ name: element.label, value: element.relPoints });
```

Comment:
```suggestion
                weight.series.push({ name: label, value: element.relWeight, bonus: element.relScore });
                weightAndBonus.series.push({ name: label, value: element.relScore, weight: element.relWeight });

                points.series.push({ name: label, value: element.relPoints });
```
`const label = element.label` in line 163

Reply:
Changed in f29991a